File: update_dates.js
const fs = require('fs');
const path = require('path');

const today = "2026-05-04";
const pattern = /lastUpdate="[^"]*"/g;

function walk(dir) {
    const files = fs.readdirSync(dir);
    files.forEach(file => {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            walk(fullPath);
        } else if (file.endsWith('.tsx')) {
            const content = fs.readFileSync(fullPath, 'utf8');
            const newContent = content.replace(pattern, `lastUpdate="${today}"`);
            if (content !== newContent) {
                fs.writeFileSync(fullPath, newContent, 'utf8');
                console.log(`Updated: ${fullPath}`);
            }
        }
    });
}

walk('app');

File: update_dates.py
import os
import re

today = "2026-05-04"
# Pattern to match both lastUpdate="YYYY-MM-DD" and lastUpdate="DD/MM/YYYY"
pattern = re.compile(r'lastUpdate="[^"]*"')

for root, dirs, files in os.walk("app"):
    for file in files:
        if file.endswith(".tsx"):
            path = os.path.join(root, file)
            try:
                with open(path, "r", encoding="utf-8") as f:
                    content = f.read()
                
                new_content = pattern.sub(f'lastUpdate="{today}"', content)
                
                if content != new_content:
                    with open(path, "w", encoding="utf-8") as f:
                        f.write(new_content)
                    print(f"Updated: {path}")
            except Exception as e:
                print(f"Error updating {path}: {e}")
